lib/matplotlib/backend_tools.py

```diff
@@ -0,0 +1,927 @@
+
+    def refresh_locators(self):
+        """Redraw the canvases, update the locators"""
+        for a in self.figure.get_axes():
+            xaxis = getattr(a, 'xaxis', None)
+            yaxis = getattr(a, 'yaxis', None)
+            zaxis = getattr(a, 'zaxis', None)
+            locators = []
+            if xaxis is not None:
+                locators.append(xaxis.get_major_locator())
+                locators.append(xaxis.get_minor_locator())
+            if yaxis is not None:
+                locators.append(yaxis.get_major_locator())
+                locators.append(yaxis.get_minor_locator())
+            if zaxis is not None:
+                locators.append(zaxis.get_major_locator())
+                locators.append(zaxis.get_minor_locator())
+
+            for loc in locators:
+                loc.refresh()
+        self.figure.canvas.draw_idle()
+
+    def home(self):
+        """Recall the first view and position from the stack"""
+        self.views[self.figure].home()
+        self.positions[self.figure].home()
+
+    def back(self):
+        """Back one step in the stack of views and positions"""
+        self.views[self.figure].back()
+        self.positions[self.figure].back()
+
+    def forward(self):
+        """Forward one step in the stack of views and positions"""
+        self.views[self.figure].forward()
+        self.positions[self.figure].forward()
+
+
+class ViewsPositionsBase(ToolBase):
+    """Base class for `ToolHome`, `ToolBack` and `ToolForward`"""
+
+    _on_trigger = None
+
+    def trigger(self, sender, event, data=None):
+        self.navigation.get_tool('viewpos').add_figure()
+        getattr(self.navigation.get_tool('viewpos'), self._on_trigger)()
+        self.navigation.get_tool('viewpos').update_view()
+
+
+class ToolHome(ViewsPositionsBase):
+    """Restore the original view lim"""
+
+    description = 'Reset original view'
+    image = 'home.png'
+    keymap = rcParams['keymap.home']
+    _on_trigger = 'home'
+
+
+class ToolBack(ViewsPositionsBase):
+    """Move back up the view lim stack"""
+
+    description = 'Back to  previous view'
+    image = 'back.png'
+    keymap = rcParams['keymap.back']
+    _on_trigger = 'back'
+
+
+class ToolForward(ViewsPositionsBase):
+    """Move forward in the view lim stack"""
+
+    description = 'Forward to next view'
+    image = 'forward.png'
+    keymap = rcParams['keymap.forward']
+    _on_trigger = 'forward'
+
+
+class ConfigureSubplotsBase(ToolBase):
+    """Base tool for the configuration of subplots"""
+
+    description = 'Configure subplots'
+    image = 'subplots.png'
+
+
+class SaveFigureBase(ToolBase):
+    """Base tool for figure saving"""
+
+    description = 'Save the figure'
+    image = 'filesave.png'
+    keymap = rcParams['keymap.save']
+
+
+class ZoomPanBase(ToolToggleBase):
+    """Base class for `ToolZoom` and `ToolPan`"""
+    def __init__(self, *args):
+        ToolToggleBase.__init__(self, *args)
```

Comment:
I really don't know how healthy is to mix super and and direct method calling....